src/gwt/src/org/rstudio/studio/client/application/ApplicationInterrupt.java

```diff
@@ -108,6 +116,61 @@ public void onFailure()
       }
    }
    
+   public void interruptRNoDebug(final InterruptHandler handler) {
+      final int originalDebugType = errorManager_.getErrorHandlerType();
+      
+      if (originalDebugType != ErrorHandlerType.ERRORS_BREAK) {
+         interruptR(handler);
+      }
+      else {
+         errorManager_.setDebugSessionHandlerType(
+            ErrorHandlerType.ERRORS_MESSAGE,
+            new ServerRequestCallback<Void>()
+            {
+               @Override
+               public void onResponseReceived(Void v)
+               {
+                  interruptR(new InterruptHandler() {
+                     @Override
+                     public void onInterruptFinished()
+                     {
+                        errorManager_.setDebugSessionHandlerType(
+                           originalDebugType,
+                           new ServerRequestCallback<Void>()
+                           {
+                              @Override
+                              public void onResponseReceived(Void v)
+                              {
+                                 interruptR(new InterruptHandler() {
```

Comment:
There should not since the RPC is not going back to R code, to my knowledge. It eventually calls: https://github.com/rstudio/rstudio/blob/54cd3abcfc58837b433464c793fe9b03a87f0bb4/src/cpp/session/modules/SessionErrors.cpp#L70-L82 which seems R-free to this point.